backend/src/utils/mod.rs

```diff
@@ -456,3 +453,94 @@ impl From<ShortChannelID> for u64 {
         id.0
     }
 }
+
+impl std::str::FromStr for PaymentState {
+    type Err = String;
+    
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "inflight" | "in_flight" => Ok(PaymentState::Inflight),
```

Comment:
"in_flight" is accepted only when parsing user or external input, but all internal code and output use "inflight" as the standard representation. why are are not  using both further in the code.